packages/react-portal-compat/src/PortalCompatProvider.tsx

```diff
@@ -0,0 +1,34 @@
+import * as React from 'react';
+import { useThemeClassName } from '@fluentui/react-components';
+
+import { PortalCompatContext } from './PortalCompatContext';
+import { RegisterPortalFn } from './types';
+
+export const PortalCompatProvider: React.FC = props => {
+  const { children } = props;
+
+  const themeClassName = useThemeClassName();
+  const cssVariablesClassName = React.useMemo<string | undefined>(
+    // "themeClassName" may contain multiple classes while we want to add only a class that hosts CSS variables
+    // Keep in sync with "packages/react-provider/src/components/FluentProvider/useThemeStyleTag.ts"
+    () => themeClassName.match(/(fui-FluentProvider\d+)/)?.[1],
```

Comment:
**suggestion:** can we move the regexp to root scope so it's instantiated only once ?

Reply:
Moved out, now we have `CLASS_NAME_REGEX` const